### PR TITLE
AMP-29221: Update the api error mechanism

### PR DESCRIPTION
**Pull request by vchihai**

### CODE REVIEW

File: amp/pom.xml [hunk 1]
```diff
@@ -1600,6 +1600,27 @@
...
+                <executions>
+                    <execution>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>java</goal>
+                        </goals>
+                        <configuration>
+                            <mainClass>org.dgfoundation.amp.asciidoc.ApiErrorAsciiDocCreator</mainClass>
+                            <arguments>
+                                <argument>${project.basedir}/src/main/asciidoc</argument>
```
Comment (developster) [inline amp/pom.xml]: Temporary apierrors.doc is generated and written in source folders. For such purposes we should use target folder (or better ${project.build.directory}). After compile phase I can see this file as unstaged in git.

File: amp/pom.xml [hunk 2]
```diff
@@ -1600,6 +1600,27 @@
                 </executions>
             </plugin>
 
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>exec-maven-plugin</artifactId>
+                <version>1.1.1</version>
+                <executions>
+                    <execution>
+                        <phase>compile</phase>
```
Comment (developster) [inline amp/pom.xml]: Please name this run. It is much easier to understand build output that way.
Ex: `<id>Generate API Errors</id>`

File: amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/GenericErrors.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.digijava.kernel.ampapi.endpoints.errors;
+
+public final class GenericErrors {
+    
+    public static final ApiErrorMessage UNKNOWN_ERROR = new ApiErrorMessage(0, 0, "Unknown Error");
```
[the remark this answers is not in the document]
Comment (vchihai) [reply amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/GenericErrors.java]: yes, agree, we will use only this type of constructor, with id and typeId

File: amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/ApiErrorMessage.java [hunk 4]
```diff
@@ -36,37 +41,63 @@
...
     }
 
     /**
      * Defines an ApiErrorMessahe
      * @param id see {@link #id id}
      * @param description see {@link #description}
      */
-    public ApiErrorMessage(Integer id, String description) {
-        this(id, description, null, null, false);
+    public ApiErrorMessage(int id, String description) {
```
[the remark this answers is not in the document]
Comment (vchihai) [reply amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/ApiErrorMessage.java]: We will remove this constructor

File: amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/ApiErrorMessage.java [hunk 5]
```diff
@@ -36,37 +41,63 @@
...
 
     /**
      * Defines an ApiErrorMessage
      * @param id see {@link #id}
      * @param description see {@link #description}
      * @param prefix see {@link #prefix}
      */
-    public ApiErrorMessage(Integer id, String description, String prefix) {
-        this(id, description, prefix, null, false);
+    public ApiErrorMessage(int id, String description, String prefix) {
```
[the remark this answers is not in the document]
Comment (vchihai) [reply amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/errors/ApiErrorMessage.java]: The constructor will be removed.

**Pull request merged**